doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,597 @@
+contributes to all output pixels.
+
+In general, a convolution described by
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s = 1`,
+* padding size :math:`p = 0`
+
+has a transpose described by
+
+* input size :math:`i' = i - (k - 1)`,
+* kernel size :math:`k' = k`,
+* stride :math:`s' = 1`,
+* padding size :math:`p' = k - 1`.
+
+Note that this corresponds to a "fully padded" convolution:
+
+.. math::
+
+    \begin{split}
+    o &= i' + 2(k - 1) - (k - 1) \\
+      &= i - (k - 1) + (k - 1) \\
+      &= i.
+    \end{split}
+
+Zero padding, transposed
+========================
+
+If the transpose of a non-padded convolution involves adding zero-padding, what
+does the transpose of a zero-padded convolution look like? It turns out that it
+involves *removing* some padding, i.e. a convolution described by
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s = 1`,
+* padding size :math:`p`
+
+has a transpose described by
+
+* input size :math:`i' = i + 2p - (k - 1)`,
+* kernel size :math:`k' = k`,
+* stride :math:`s' = 1`,
+* padding size :math:`p' = k - p - 1`.
+
+The output size is
+
+.. math::
+
+    \begin{split}
+    o &= i' + 2(k - p - 1) - (k - 1) \\
+      &= i + 2p - (k - 1) - 2p + 2(k - 1) - (k - 1) \\
+      &= i.
+    \end{split}
+
+Here is what the transpose of a convolution looks like for :math:`i = 5`,
+:math:`k = 4` and :math:`p = 2`:
+
+.. figure:: conv_arithmetic_figures/arbitrary_padding_no_strides_transposed.gif
+
+Special cases
+-------------
+
+Half
+^^^^
+
+Unsurprisingly, the transpose of a "half padded" convolution is itself a "half
+padded" convolution, i.e. a convolution described by
+
+* input size :math:`i`,
+* kernel size :math:`k = 2n + 1`,
+* stride :math:`s = 1`,
+* padding size :math:`p = \lfloor k / 2 \rfloor = n`
+
+has a transpose described by
+
+* input size :math:`i' = i`,
+* kernel size :math:`k' = k`,
+* stride :math:`s' = 1`,
+* padding size :math:`p' = p`.
+
+Here is what the transpose of a convolution looks like for :math:`i = 5`,
+:math:`k = 3` and :math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/same_padding_no_strides_transposed.gif
+
+Full
+^^^^
+
+If the transpose of a convolution with no padding is a "fully padded"
+convolution, then the transpose of a "fully padded" convolution is... a
+convolution with no padding. A convolution described by
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s = 1`,
+* padding size :math:`p = k - 1`
+
+has a transpose described by
+
+* input size :math:`i' = i + (k - 1)`,
+* kernel size :math:`k' = k`,
+* stride :math:`s' = 1`,
+* padding size :math:`p' = 0`.
+
+Here is what the transpose of a convolution looks like for :math:`i = 5`,
+:math:`k = 3` and :math:`p = 2`:
+
+.. figure:: conv_arithmetic_figures/full_padding_no_strides_transposed.gif
+
+Strides, transposed
+===================
+
+If a zero-padded convolution has a transpose with padding removed, then a
```

Comment:
I suggest "Analogously to the zero padding case, if a zero-padded convolution corresponds to a cropped transpose convolution, then a convolution with strides greater than 1 has a corresponding transpose with strides [...]"